tests/Unit/PointwiseFunctions/Elasticity/Test_ElasticPotentialEnergy.cpp

```diff
@@ -0,0 +1,81 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/Elasticity/Tags.hpp"  // IWYU pragma: keep
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/IsotropicHomogeneous.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/Tags.hpp"
+#include "PointwiseFunctions/Elasticity/ElasticPotentialEnergy.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+
+template <size_t Dim>
+Scalar<DataVector> potential_energy_density(
+    const tnsr::ii<DataVector, Dim, Frame::Inertial>& strain,
+    const tnsr::I<DataVector, Dim>& coordinates, const double bulk_modulus,
+    const double shear_modulus) noexcept {
+  Elasticity::ConstitutiveRelations::IsotropicHomogeneous<Dim>
+      constitutive_relation{bulk_modulus, shear_modulus};
+  return Elasticity::potential_energy_density<Dim>(
+      strain, coordinates, std::move(constitutive_relation));
+}
+
+template <size_t Dim>
+void test_elastic_potential_energy(const DataVector& used_for_size) {
+  pypp::check_with_random_values<4>(
+      &potential_energy_density<Dim>, "Elasticity.ElasticPotentialEnergy",
+      "potential_energy_density", {{{-1., 1.}, {0., 1.}, {0., 1.}, {0., 1.}}},
+      used_for_size);
+}
+
+template <size_t Dim>
+void test_compute_tags(const DataVector& used_for_size) noexcept {
+  using strain_tag = Elasticity::Tags::Strain<Dim>;
+  using energy_tag = Elasticity::Tags::PotentialEnergy<Dim>;
+  using energy_compute_tag = Elasticity::Tags::PotentialEnergyCompute<Dim>;
+  using constitutive_relation_tag = Elasticity::Tags::ConstitutiveRelation<
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<Dim>>;
+  using coordinates_tag = domain::Tags::Coordinates<Dim, Frame::Inertial>;
+  const size_t num_points = used_for_size.size();
+  {
+    INFO("Energy" << Dim << "D");
```

Comment:
Instead of the `<< Dim << "D"` you can add `CAPTURE(Dim);` somewhere, e.g. at the start of the function